### ISSUE TITLE
Bugs: Incorrect type of exported UserConfig

### ISSUE THREAD

**Issue by TuiMao233** [YunYouJun/valaxy#28]
```ts
import type { UserConfig } from 'valaxy'
import type { ThemeUserConfig } from 'valaxy-theme-yun'

const config: UserConfig<ThemeUserConfig> = {/* ... */}

// any
config.aaa
// any
config.theme

export default config
```

**Comment by YunYouJun** [comment]
Which version?

**Comment by TuiMao233** [comment]
> Which version?

valaxy - 0.7.7
valaxy-theme-yun - 0.7.7
pnpm - 7.1.2
node - 16.15.0
system - window

**Comment by YunYouJun** [comment]
Try this:

```ts
import { defineConfig } from 'valaxy'
import type { ThemeConfig } from 'valaxy-theme-yun'

export default defineConfig<ThemeConfig>({
  // ...
})
```

**Comment by TuiMao233** [comment]
> Try this:
> 
> ```ts
> import { defineConfig } from 'valaxy'
> import type { ThemeConfig } from 'valaxy-theme-yun'
> 
> export default defineConfig<ThemeConfig>({
>   // ...
> })
> ```

defineConfig is correct

**Comment by TuiMao233** [comment]
I tried to read the property from config, it is still any, this problem is also mentioned in #34

defineConfig (here), can get type hints normally, but the external config is abnormal

![image](https://user-images.githubusercontent.com/49724027/178113310-988037e0-fd1a-4667-be1c-d4d5f2ce96ca.png)

**Comment by TuiMao233** [comment]
Not only has this problem with defineConifg, but also with `useConfig`
![image](https://user-images.githubusercontent.com/49724027/178114643-cace0b1a-58cc-4513-bf56-103a5fab9cee.png)

**Comment by YunYouJun** [comment]
I think this is because our theme is runtime TS, so it references the Types TS file, but I only publish it compiled.

We can indeed change to 'valaxy/types' and then I will add an export of'./types' to 'package.json'.

**Comment by YunYouJun** [comment]
I think it is fixed in `v0.7.8`.

**Comment by YunYouJun** [comment]
There still seems to be some compilation issues, I will fix it as soon as possible.

**Comment by YunYouJun** [comment]
I think it had been fixed in `v0.9.0`.
